### PR TITLE
Stop running Arch in CI, and update test matrix

### PR DESCRIPTION
**Pull request by eskaur**

### CODE REVIEW

**Review by trym-b — DISMISSED** [review]
Added one comment; otherwise looks good to me.

File: .github/workflows/main.yml
Comment (nedrebo) [inline .github/workflows/main.yml:98]: Is it hard to change this to the following?

```suggestion
    # - 4 latest Ubuntu versions
```

The reason I ask is that it is highly likely that we got users on any recent Ubuntu, we do not want to leave them behind immediately when a new LTS is out.

File: .github/workflows/main.yml
Comment (nedrebo) [inline .github/workflows/main.yml:97]:
```suggestion
    # - All Ubuntu versions supported by Zivid SDK
```

Technically the same, but more robust formulation without the LTS.

File: .github/workflows/main.yml
Comment (eskaur) [reply .github/workflows/main.yml:98]: I'll try. Could be easy.

File: .github/workflows/main.yml
Comment (eskaur) [reply .github/workflows/main.yml:101]: Isn't it still true that the installer still only officially supports 16.04? I haven't changed out the installer.

File: .github/workflows/main.yml
Comment (eskaur) [reply .github/workflows/main.yml:98]: Wait Ubuntu 21.04 and 21.10 are already past end-of-life. The apt repos aren't even up anymore.

Comment (nedrebo) [reply .github/workflows/main.yml:98]: Ok, resolving this thread then.

File: .github/workflows/main.yml
Comment (nedrebo) [reply .github/workflows/main.yml:97]: This is a minor comment. Approving now, fix before merging is you agree.

**Review by nedrebo — APPROVED** [review]

File: .github/workflows/main.yml
Comment (trym-b) [reply .github/workflows/main.yml:101]: Ok, then let's consider it as a follow-up. Might be we do not want to touch this at all

File: .github/workflows/main.yml
Comment (eskaur) [reply .github/workflows/main.yml]: Done

**Review by trym-b — APPROVED** [review]

**Pull request merged**
